### PR TITLE
PARQUET-2386: More consistent code style in parquet-mr

### PR DESCRIPTION
**Pull request by amousavigourabi**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET/) issues and references them in the PR title. For example, "PARQUET-1234: My Parquet PR"
  - https://issues.apache.org/jira/browse/PARQUET-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
    - This PR only refactors style, no logic is added or removed in any way shape or form

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does
  - Adds note in the PR template on the style checks

---

This PR contains two commits, the first adds the style checks and configurations, the second applies these changes to the repository.

### CODE REVIEW

**Comment by amousavigourabi** [comment]
@wgtmac

**Comment by wgtmac** [comment]
Thanks for the improvement!

Could you please take a look at this? @shangxinli @gszadovszky @Fokko

File: pom.xml
Comment (gszadovszky) [inline pom.xml:538]: Does it mean we allow tabs for indention? (Personally, I don't like tabs because it might be rendered differently on different systems.)

**Review by gszadovszky — COMMENTED** [review]
Thank you @amousavigourabi for working on this!
One comment about tabs for indention, otherwise LGTM.

**Comment by Fokko** [comment]
This is great @amousavigourabi. I was thinking of adding linting as well, so great work here!

File: pom.xml
Comment (amousavigourabi) [reply pom.xml:538]: No, tabs are _not_ allowed by this configuration. This block is a way to make the linter work as it should with two space indents as used by Parquet, instead of four.

Comment (amousavigourabi) [reply pom.xml:538]: It is a bit hacky, but it also allows contributors to run the `spotless:apply` goal in case they are for whatever reason using tabs in their development environment.

**Comment by amousavigourabi** [comment]
@Fokko , how do we want to coordinate this pull request and your refactoring PRs? I do not feel a lot for applying the editorconfig and Spotless a dozen or so times to resolve conflicts in this PR after each of those get merged.

**Comment by amousavigourabi** [comment]
I rebased on the latest master. I'd rather get this merged somewhat quickly (maybe by the end of the week?) as to avoid blocking other merges and/or endless rebasing.

**Review by gszadovszky — APPROVED** [review]

**Review by wgtmac — APPROVED** [review]

**Comment by wgtmac** [comment]
> I rebased on the latest master. I'd rather get this merged somewhat quickly (maybe by the end of the week?) as to avoid blocking other merges and/or endless rebasing.

I just merged this PR to unblock others. Thanks @amousavigourabi!

**Pull request merged**
